### ISSUE TITLE
 V22.03.0 release currently does not work on D-Link DIR-645 

### ISSUE THREAD

**Issue by user8446** [openwrt/openwrt#10634]
[D-Link DIR-645](https://openwrt.org/toh/d-link/dir-645)

I did a sysupgrade from v21.02.3 and the LAN is dead. I then tried a factory v22.03.0 image through the built-in D-Link recovery to eliminate a sysupgrade issue and the LAN is still dead.

I went back to v21.02.3 through recovery and the router is back up.

X-Posted to the forums: https://forum.openwrt.org/t/v22-03-0-release-does-not-work-on-d-link-dir-645/136435 and put a temp note on the [DIR-645 wiki](https://openwrt.org/toh/d-link/dir-645)

**Comment by user8446** [comment]
I was able to connect via serial and here is the output. The following just repeats over and over:

```
U-Boot 1.1.3 (Apr 15 2011 - 10:39:36) (ALPHA)
SVN revision: 516
Target board: WRG-N39

Board: Ralink APSoC DRAM:  64 MB
spi_wait_nsec: 30
spi device id: c2 20 17 c2 20 (2017c220)
find flash: MX25L6405D
raspi_read: from:30000 len:1000
.raspi_read: from:30000 len:1000
.============================================
Ralink UBoot Version: 3.3
--------------------------------------------
ASIC 3883_MP (MAC to REALTEK Mode)
DRAM component: 512 Mbits DDR, width 16
DRAM bus: 16 bit
Total memory: 64 MBytes
Flash component: SPI Flash
Date:Apr 15 2011  Time:10:39:36
============================================
icache: sets:512, ways:4, linesz:32 ,total:65536
dcache: sets:256, ways:4, linesz:32 ,total:32768
gpio_init: write 0x00001801
gpio_init: write 0x0000185d to GPIOMODE(0xb0000060)

 Initializing RTL8367 GbE switch ...
 Setting RTL8367RB RGMII Delay time tx:1,rx0

Please choose the operation:
   1: Load system code to SDRAM via TFTP.
   2: Load system code then write to Flash via TFTP.
   3: Boot system code via Flash (default).
   4: Entr boot command line interface.
   7: Load Boot Loader code then write to Flash via Serial.
   9: Load Boot Loader code then write to Flash via TFTP.                                                                                                                                  0

3: System Boot system code via Flash.
## Booting image at bc050000 ...
raspi_read: from:50000 len:40
.raspi_read: from:50000 len:c
.raspi_read: from:50000 len:5738a6
........................................................................................We have SEAMA, Image Size = 5716070
Verifying Checksum ...
Uncompressing SEAMA linux.lzma ... LZMA ERROR 1 - must RESET board to recover

```

**Comment by xabolcs** [comment]
For the record: with the help of @user8446, I was able to create a PR which fixes the problem: #10834

**Comment by xabolcs** [comment]
> **user8446** closed this as 5 days ago

While the fix is ready, it's still not included in **master** not even on **openwrt-22.03**.

May I ask to reopen the issue?

**Comment by DragonBluep** [comment]
This issue and https://github.com/openwrt/openwrt/issues/6450 were fixed by https://github.com/openwrt/openwrt/commit/fafc9448aec907d40022fea0c3c75ad906b67d3e (v22.03) and https://github.com/openwrt/openwrt/commit/c293b492dfa114b67e90d5434edfeba17ba29980 (main), can be closed.

Ping @robimarko
